### ISSUE TITLE
Spotify Recently Listened always returns 400

### ISSUE THREAD

**Issue by micahlt** [lowlighter/metrics#24]
**Bug description**
When the Recently Listened Tracks plugin is used, a message is displayed that claims that the API request received a 400 HTTP response.  I've followed the setup tutorials multiple times and nothing fixes it.  

**Expected behavior**
Displays recently listened tracks.

**To Reproduce**
Follow the tutorial to setup the Recently Listened plugin.  

**Additional context and screenshots**
Here are the last lines of the GitHub Actions output: 
```bash
/usr/bin/docker run --name cc4956019579e8637340a187fb341d5012884c_b8f32e --label cc4956 --workdir /github/workspace --rm -e INPUT_TOKEN -e INPUT_COMMITTER_TOKEN -e INPUT_USER -e INPUT_TEMPLATE -e INPUT_BASE -e INPUT_PLUGIN_MUSIC -e INPUT_PLUGIN_MUSIC_PROVIDER -e INPUT_PLUGIN_MUSIC_TOKEN -e INPUT_PLUGIN_MUSIC_MODE -e INPUT_PLUGIN_MUSIC_LIMIT -e INPUT_FILENAME -e INPUT_OPTIMIZE -e INPUT_REPOSITORIES -e INPUT_PLUGIN_PAGESPEED -e INPUT_PLUGIN_PAGESPEED_DETAILED -e INPUT_PLUGIN_PAGESPEED_TOKEN -e INPUT_PLUGIN_LINES -e INPUT_PLUGIN_TRAFFIC -e INPUT_PLUGIN_HABITS -e INPUT_PLUGIN_HABITS_FROM -e INPUT_PLUGIN_LANGUAGES -e INPUT_PLUGIN_LANGUAGES_IGNORED -e INPUT_PLUGIN_LANGUAGES_SKIPPED -e INPUT_PLUGIN_FOLLOWUP -e INPUT_PLUGIN_MUSIC_PLAYLIST -e INPUT_PLUGIN_POSTS -e INPUT_PLUGIN_POSTS_SOURCE -e INPUT_PLUGIN_POSTS_LIMIT -e INPUT_PLUGIN_ISOCALENDAR -e INPUT_PLUGIN_ISOCALENDAR_DURATION -e INPUT_PLUGIN_GISTS -e INPUT_PLUGIN_TOPICS -e INPUT_PLUGIN_TOPICS_SORT -e INPUT_PLUGIN_TOPICS_LIMIT -e INPUT_PLUGIN_PROJECTS -e INPUT_PLUGIN_PROJECTS_LIMIT -e INPUT_DEBUG -e INPUT_VERIFY -e INPUT_DRYRUN -e INPUT_PLUGINS_ERRORS_FATAL -e HOME -e GITHUB_JOB -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_REPOSITORY_OWNER -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_RETENTION_DAYS -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_SERVER_URL -e GITHUB_API_URL -e GITHUB_GRAPHQL_URL -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e GITHUB_ACTION_REPOSITORY -e GITHUB_ACTION_REF -e GITHUB_PATH -e GITHUB_ENV -e RUNNER_OS -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e ACTIONS_CACHE_URL -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/micahlt/micahlt":"/github/workspace" cc4956:019579e8637340a187fb341d5012884c
GitHub metrics
========================================================
Version                   | 2.8.0
Configuration             | loaded
Template to use           | classic
Github token              | provided
Github GraphQL API        | ok
Github REST API           | ok
SVG output file           | github-metrics.svg
SVG optimization          | true
GitHub user               | micahlt
Debug mode                | false
Base parts                | header, activity, community, repositories, metadata
Plugins enabled           | music
Music provider            | spotify
Music plugin mode         | recent
Music playlist            | null
Music tracks limit        | 4
Music token               | provided
Repositories to use       | 100
Plugin errors             | ignore
Render                    | complete
Verify SVG                | false
Repository                | micahlt/micahlt
Branch                    | master
Committer token           | provided
Committer REST API        | ok
Committer                 | (unknown)
Previous render sha       | 03e46c0bb92285b2c5783d9e48c0c1769f5ac7cc
Commit to repo            | ok
Success !
```

**Comment by lowlighter** [comment]
Hi, 
Thanks for reaching out !

I would say that it probably comes from your spotify tokens.
 
Unfortunately, it's not possible yet to see the full error message of the API request (which is stored in response body), but I'll change it in a future release for easier debug.

So what you could do for now, is to go on spotify.com from your browser and execute the following JavaScript in the web console (replacing `***` with your tokens) : 
```js
(async () => {
  const req = await fetch("https://accounts.spotify.com/api/token", {
    method:"post",
    headers:{"Content-Type":"application/x-www-form-urlencoded"},
    body:`${new URLSearchParams({grant_type:"refresh_token", refresh_token:"***", client_id:"***", client_secret:"***"})}`,
  })
  console.log(req.status, await req.json())
})()
```

Basically, it'll make the same API request used by the music plugin to retrieve a new token.

From what I tested, it'll say which field is incorrect and you'll be able to view the whole error message : 
```
400 Object { error: "invalid_grant", error_description: "Invalid refresh token" }
400 Object { error: "invalid_client", error_description: "Invalid client" }
400 Object { error: "invalid_client", error_description: "Invalid client secret" }
```

I assume it's the same for everyone, but for record my `client_id`/`client_secret` have a 32-chars length while my `refresh_token` has a 131-chars length, so you could also check that you're not missing a character in them. Maybe try to swap your `client_id`/`client_secret`, in case they have been accidently inverted.

Have a nice day 👍
